nwind/src/unwind_context.rs

```diff
@@ -96,6 +96,12 @@ impl< 'a, A: Architecture > UnwindHandle< 'a, A > {
 
         self.ctx.nth_frame += 1;
 
+        // avoid infinite loops
+        if self.ctx.nth_frame > 1000 {
+            warn!("possible infinite loop detected and avoided");
+            return false;
+        }
```

Comment:
Have you actually hit a genuine infinite loop here? AFAIK infinite loops shouldn't really be possible as you're going to overflow the stack sooner or later anyway.

Anyway, this change isn't really correct. Even though > 1000 frame deep stacks are certainly a sign of a problem they should still be gathered. I've seen such stack traces in the wild, and gathering as much of it as possible later helps to fix it if you can manage to get to the top. So what was your motivation in adding this here? If you want to limit stack traces to a certain length we could add an extra parameter instead.

Reply:
Yes, I've got it frequently in one of the cortex-15 app, but I cannot post it here nor dig into it further since I'm leaving the company.

What I managed to figure out through is that it looked like a arm unwinder bug, vec holding the frames kept allocating until it failed while trying to reallocate into 3.5 gigs.

Reply:
Hi, do you still want to make this as command line option? I'm asking because failed allocation, which is how this bug manifests itself, is just an abort, no panic nor Err. This makes it hard to diagnose if it happens to someone.